src/ray/raylet/scheduling_queue.h

```diff
@@ -151,21 +151,62 @@ class SchedulingQueue {
   /// \param filter_state The task state to filter out.
   void FilterState(std::unordered_set<TaskID> &task_ids, TaskState filter_state) const;
 
+  class TaskQueue {
+   public:
+    /// Creating a task queue.
+    TaskQueue() {}
+
+    /// Destructor for task queue.
+    ~TaskQueue();
+
+    /// \brief Append a task to queue.
+    ///
+    /// \param task_id The task ID for the task to append.
+    /// \param task The task to append to the queue.
```

Comment:
for all methods with a return value, please also document the return value using `\return`